django_rq/queues.py

```diff
@@ -107,12 +107,30 @@ def get_redis_connection(config, use_strict_redis=False):
     return redis_cls(host=config['HOST'], port=config['PORT'], db=config['DB'], password=config.get('PASSWORD', None))
 
 
+# Used as cache to store a Redis client for each queue
+REDIS_CLIENT_BY_QUEUE = {}
+
+
+def get_cached_client_by_queue(name, use_strict_redis=False):
+    """
+    Fetch a Redis client from REDIS_CLIENT_BY_QUEUE. Call get_redis_connection to instantiate a new one if client
+    does not exist yet.
+    """
+    global REDIS_CLIENT_BY_QUEUE
+    from .settings import QUEUES
+    try:
+        client = REDIS_CLIENT_BY_QUEUE[name]
+    except KeyError:
+        client = get_redis_connection(QUEUES[name], use_strict_redis)
```

Comment:
@yuxabc It looks like you will create  different clients with different behavior depending on the value of use_strict_redis use_strict_redis. This means that the key to your dict cannot be name, but tuple(name, use_strict_redis).